openlibrary/catalog/get_ia.py

```diff
@@ -46,7 +47,7 @@ def bad_ia_xml(identifier):
     # need to handle 404s:
     # http://www.archive.org/details/index1858mary
     loc = "{0}/{0}_marc.xml".format(identifier)
-    return '<!--' in urlopen_keep_trying(IA_DOWNLOAD_URL + loc).read()
+    return '<!--' in urlopen_keep_trying(IA_DOWNLOAD_URL + loc).text
```

Comment:
HTTPResponse.read() is going to return `bytes` so normally we would want `requests.get().content` (also bytes) but here I think that you have it right...  We want `requests.get().text` (also str)

Reply:
Agreed.